_internal/utils/preload.ts

```diff
@@ -0,0 +1,30 @@
+import { Middleware, Key, BareFetcher } from '../types'
+import { serialize } from './serialize'
+
+const REQUEST = new Map<string, Promise<any>>()
```

Comment:
should this be bound with cache in global state?

Reply:
@huozhi Thank you!　That makes sense to me.
But does the preload function have to accept `cache` as the argument to use the global state? (It might work if we always use a **global** cache to store the value)

```jsx
// we need the cache argument to get REQUEST from SWRGlobalState
preload(key, fetcher, cache) 
```

Reply:
I've used `SWRGlobalState` to store preloading requests.
https://github.com/vercel/swr/pull/2026/commits/85f160cdafebf2e88d69a72ee5b1b4503ce39681